src/hooks/useValidateTransactionMemo.ts

```diff
@@ -0,0 +1,136 @@
+import { TransactionBuilder } from "@stellar/stellar-sdk";
+import {
+  mapNetworkToNetworkDetails,
+  STORAGE_KEYS,
+  TRANSACTION_WARNING,
+} from "config/constants";
+import { MemoRequiredAccountsApiResponse } from "config/types";
+import { useAuthenticationStore } from "ducks/auth";
+import { usePreferencesStore } from "ducks/preferences";
+import { useTransactionBuilderStore } from "ducks/transactionBuilder";
+import { useTransactionSettingsStore } from "ducks/transactionSettings";
+import { cachedFetch } from "helpers/cachedFetch";
+import { isMainnet } from "helpers/networks";
+import { getApiStellarExpertIsMemoRequiredListUrl } from "helpers/stellarExpert";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { stellarSdkServer } from "services/stellar";
+
+export const useValidateTransactionMemo = () => {
+  const { network } = useAuthenticationStore();
+  const { transactionXDR } = useTransactionBuilderStore();
```

Comment:
I wonder if we should pass `transactionXDR` as an input for `useValidateTransactionMemo` like `useValidateTransactionMemo(transactionXDR)` instead of retrieving it from `useTransactionBuilderStore` since for the dApp transaction case (WalletKit) it does not use the `useTransactionBuilderStore` at all. If you look at `approveSessionRequest` in `walletKitUtil.ts` you will see that the `xdr` in this case is being retrieved directly from the `sessionRequest` event.

Are you able to currently test this Sign dApp transaction flow using StellarTerm as suggested [on the issue](https://github.com/stellar/freighter-mobile/issues/248)? Let me know in case you need any help with it, you'll need to set some `.env` vars on the project in order to test the Wallet Connect (WalletKit) flow